src/pages/fi-pv-expenses/forms/PaymentVoucherExpensesForm.js

```diff
@@ -0,0 +1,739 @@
+                    }}
+                    error={formik.touched.templateId && Boolean(formik.errors.templateId)}
+                    maxAccess={maxAccess}
+                />
+            </Grid>
+            <Grid item xs={3}>
+                <CustomTextField
+                    name='amount'
+                    label={labels.amount}
+                    value={amountSum}
+                    readOnly
+                    maxAccess={maxAccess}
+                    onChange={formik.handleChange}
+                    onClear={() => formik.setFieldValue('amount', '')}
+                    error={formik.touched.amount && Boolean(formik.errors.amount)}
+                />
+            </Grid>
+            <Grid item xs={6}>
+                <ResourceLookup
+                    endpointId={CashBankRepository.CashAccount.snapshot}
+                    parameters={{
+                      _type: 0
+                    }}
+                    name='cashAccountRef'
+                    readOnly={isPosted || isCancelled}
+                    label={labels.cash}
+                    valueField='reference'
+                    displayField='name'
+                    valueShow='cashAccountRef'
+                    secondValueShow='cashAccountName'
+                    form={formik}
+                    required
+                    onChange={(event, newValue) => {
+                      formik.setValues({
+                          ...formik.values,
+                          cashAccountId: newValue?.recordId || '',
+                          cashAccountRef: newValue?.reference || '',
+                          cashAccountName: newValue?.name || ''
+                      })
+                    }}
+                    errorCheck={'cashAccountId'}
+                    maxAccess={maxAccess}
+                />
+            </Grid>
+            <Grid item xs={6}>
+                <CustomTextArea
+                    name='notes'
+                    type='text'
+                    label={labels.notes}
+                    value={formik.values.notes}
+                    readOnly={isPosted || isCancelled}
+                    rows={3}
+                    maxAccess={maxAccess}
+                    onChange={e => formik.setFieldValue('notes', e.target.value)}
+                    onClear={() => formik.setFieldValue('notes', '')}
+                    error={formik.touched.notes && Boolean(formik.errors.notes)}
+                />
+            </Grid>
+          </Grid>
+          <DataGrid
+            onChange={value => {
+              const updatedExpenses = value.map(expense => ({
+                ...expense,
+                hasCostCenters: true
+              }))
+            
+              formik.setFieldValue('expenses', updatedExpenses);
+            }}
+            value={formik?.values?.expenses}
+            error={formik?.errors?.expenses}
+            columns={columns}
+            allowDelete={true}
```

Comment:
u set allowDelete and allowAddNewLine always as true, this should behave according to isCancel or isPosted